runtime/src/main/java/org/corfudb/runtime/object/transactions/OptimisticTransactionalContext.java

```diff
@@ -111,20 +111,27 @@
                             // inside syncObjectUnsafe, depending on the object
                             // version, we may need to undo or redo
                             // committed changes, or apply forward committed changes.
-                            try {
-                                o.syncObjectUnsafe(getSnapshotTimestamp());
-                            } catch (TrimmedException te) {
-                                // If a trim is encountered, we must reset the object
-                                o.resetUnsafe();
-                                // and abort the transaction
-                                TransactionAbortedException tae =
-                                        new TransactionAbortedException(
-                                                new TxResolutionInfo(getTransactionID(),
-                                                        getSnapshotTimestamp()), null,
+                            for (int x = 0; x < this.builder.getRuntime().getTrimRetry(); x++) {
```

Comment:
Okay, this is the right place now. If you wanted code reuse, you could stuff this into a function like AbstractTransactionalContext::doTransactionalSync(VersionLockedObject object)